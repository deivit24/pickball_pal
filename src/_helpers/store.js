import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
