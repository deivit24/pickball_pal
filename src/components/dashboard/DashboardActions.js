import React from 'react';
import { useDispatch } from 'react-redux';
import { deleteAccount } from '../../_actions/profile';
import { Link } from 'react-router-dom';
import Button from '@material-ui/core/Button';
const DashboardActions = () => {
  const dispatch = useDispatch();
  return (
    <div className="row">
      <div className="col-lg-6 mb-3">
        <Link to="/edit-profile">
          <Button fullWidth variant="outlined" color="primary">
            {' '}
            <i className="fas fa-user-circle mr-1"></i> Edit Profile
          </Button>
        </Link>
      </div>
      <div className="col-lg-6 mb-3">
        <Link to="/upload-photo">
          <Button fullWidth variant="outlined" color="primary">
            {' '}
            <i className="fas fa-upload mr-1"></i> Edit Photo
          </Button>
        </Link>
      </div>
      <div className="col-lg-12 mb-3">
        <Button
          onClick={() => dispatch(deleteAccount())}
          fullWidth
          variant="outlined"
          color="secondary"
        >
          {' '}
          <i className="fas fa-trash mr-1"></i> Delete Account
        </Button>
      </div>
    </div>
  );
};

export default DashboardActions;
