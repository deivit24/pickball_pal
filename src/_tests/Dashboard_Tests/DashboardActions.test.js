import React from 'react';
import { render } from '@testing-library/react';
import DashboardActions from '../../components/dashboard/DashboardActions';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing DashboardActions

describe('Testing the DashboardActions', () => {
  it('it renders DashboardActions without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <DashboardActions />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <DashboardActions />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
