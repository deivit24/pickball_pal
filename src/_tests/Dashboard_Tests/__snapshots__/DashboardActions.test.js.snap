// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the DashboardActions matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-lg-6 mb-3"
    >
      <a
        href="/edit-profile"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             
            <i
              class="fas fa-user-circle mr-1"
            />
             Edit Profile
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
    <div
      class="col-lg-6 mb-3"
    >
      <a
        href="/upload-photo"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             
            <i
              class="fas fa-upload mr-1"
            />
             Edit Photo
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
    <div
      class="col-lg-12 mb-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
           
          <i
            class="fas fa-trash mr-1"
          />
           Delete Account
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
